src/main/java/com/online/bookstore/security/SecurityConfiguration.java

```diff
@@ -14,17 +18,28 @@
 @EnableMethodSecurity
 @RequiredArgsConstructor
 public class SecurityConfiguration {
+    @Value("${system-configuration.public-endpoints}")
+    private Set<String> publicEndpoints;
 
     @Bean
     public SecurityFilterChain securityFilterChain(final HttpSecurity http) throws Exception {
         http
                 .csrf(AbstractHttpConfigurer::disable)
-                .authorizeHttpRequests(authorizeRequests -> authorizeRequests.anyRequest()
-                        .permitAll());
+                .cors(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(authorizeRequests -> authorizeRequests
+                        .requestMatchers(publicEndpoints.toArray(String[]::new)).permitAll()
+                        .anyRequest()
+                        .authenticated())
+                .httpBasic(Customizer.withDefaults());
 
         return http.build();
     }
 
+    @Bean
+    static GrantedAuthorityDefaults grantedAuthorityDefaults() {
+        return new GrantedAuthorityDefaults("");
+    }
```

Comment:
why is it needed?

Reply:
by default spring adds ROLE_ prefix to authorities, just want to avoid adding ROLE_ to @PreAuthorize parameter

Reply:
So why it static?